ui/views.py

```diff
@@ -242,6 +242,16 @@ def extra_context(self):
 # pylint: disable=unused-argument
 # repo_slug argument will be used by the decorator to protect the view
 @login_required
+@permission_required_or_403(
```

Comment:
Since this is here, it's impossible to hit the `PermissionDenied` in the view. Since it's duplicating the same check and the decorator is a good place to do it, I think we should remove the second `except` to prevent confusion, duplication, and reduced test coverage (it should be impossible to cover that line in tests).